### PR TITLE
Changelog for 2.1.0

### PR DESCRIPTION
**Pull request by jcape**
- Fill in changelog from release checklist.

### Motivation

It's an unqualified good to tell the people who use our software what has changed.

### Future Work

- Cherry pick this and version changes to release/v2
- Release 2.1.0

### CODE REVIEW

**Comment by jcape** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2648** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2648" target="_blank">Graphite</a> 
    * **PR #2649** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2649" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2649?utm_source=stack-comment).

**Comment by jcape** [comment]
For reviewers: I'm mostly interested in whether this captures the changes in the `release/v2` branch.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,11 +8,27 @@ The crates in this repository do not adhere to [Semantic Versioning](https://sem
...
 ### Added
-- Support env overrides for ~all command-line flags.
-  - Flags that take multiple values can be repeated on the command line,
-    or passed as comma-separated values via environment or command-line args.
-- Update CI deployments to use zerossl instead of letsencrypt
+
+- Add `Chain-ID` gRPC metadata ([MCIP #49](https://github.com/mobilecoinfoundation/mcips/pull/49)) to provide additional runtime disambiguation between clients and servers.
+- Add a required `--chain-id` command-line arguments to consensus and fog servers.
+- Add an optional `--chain-id` command-line argument to `mobilecoind`.
+- Support env overrides and overlays (for multi-value items) for nearly all command-line arguments.
```
Comment (eranrund) [inline CHANGELOG.md:18]: Are overlays supported? I think one just take precedence over the other (e.g. if both env var and command line arguments for a given option are present, only the command line arguments will get used)

**Review by garbageslam — APPROVED** [review]
idk about overlays but the rest looks good

File: CHANGELOG.md (hunk 1 above)
Comment (jcape) [reply CHANGELOG.md]: Fixed

**Review by eranrund — APPROVED** [review]

**Pull request merged**